kernel/thread.c

```diff
@@ -270,6 +270,19 @@ void _setup_new_thread(struct k_thread *new_thread,
 		       void *p1, void *p2, void *p3,
 		       int prio, u32_t options)
 {
+#if CONFIG_STACK_POINTER_RANDOM
+#if defined(CONFIG_STACK_GROWS_UP)
+	/* This is so rare not bothering for now */
+#error "Stack pointer randomization not implemented for upward growing stacks"
+#endif
+	/* Don't need to worry about alignment of the size here, _new_thread()
+	 * is required to do it
+	 *
+	 * FIXME: Not the best way to get a random number in a range.
+	 * See #6493
+	 */
+	stack_size -= sys_rand32_get() % CONFIG_STACK_POINTER_RANDOM;
```

Comment:
I'm a little concerned here about this causing testing to become non-deterministic. If something only slightly overflows its stack, it may work most of the time, only failing rarely, making it hard to debug, especially without overflow checking.

Also, do we need to take into account stack alignment requirements here? Some architectures have strict alignment requirements on the stack, and others will just be less efficient with a misaligned stack.

Reply:
> testing to become non-deterministic

This isn't on by default, FWIW.

> do we need to take into account stack alignment requirements here

We do not. _new_thread() is required to do any required rounding.

Reply:
I don't think the bias is a real concern here, at least as long as sys_rand32_get() has a reasonable distribution. The worst bias is of order `1/ceil(2^32 / CONFIG_STACKPOINTER_RANDOM)`, and given that our stacks are expected to be rather small compared to the address space, I don't think is worth putting in extra code to handle removing the bias.

An extreme example is using 8-bit values, and a modulus of 170, where half of the numbers occur twice as often as others.